include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,555 @@
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator>(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right > rhs.window_right) && (lhs.num_w_elems > rhs.num_w_elems);
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator<=(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right <= rhs.window_right) && (lhs.num_w_elems <= rhs.num_w_elems);
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator>=(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right >= rhs.window_right) && (lhs.num_w_elems >= rhs.num_w_elems);
+    }
+    //!\}
+
+    //!\brief Pre-increment.
+    window_iterator & operator++() noexcept
+    {
+        get_minimiser();
+        return *this;
+    }
+
+    //!\brief Post-increment.
+    window_iterator operator++(int) noexcept
+    {
+        window_iterator tmp{*this};
+        get_minimiser();
+        return tmp;
+    }
+
+    //!\brief Return the minimiser.
+    value_type operator*() const noexcept
+    {
+        return minimiser_value;
+    }
+
+    /*!\brief Forward this iterator.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    window_iterator & operator+=(difference_type const skip) noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        get_minimiser(skip);
+        return *this;
+    }
+
+    /*!\brief Forward copy of this iterator.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    window_iterator operator+(difference_type const skip) const noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        window_iterator tmp{*this};
+        return tmp += skip;
+    }
+
+    /*!\brief Non-member operator+ delegates to non-friend operator+.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    friend window_iterator operator+(difference_type const skip, window_iterator const & it) noexcept
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        return it + skip;
+    }
+
+    /*!\brief Move the iterator by a given offset and return the corresponding minimiser.
+     * \attention This function is only avaible if `it_t` models std::random_access_iterator.
+     */
+    value_type operator[](difference_type const n)
+    //!\cond
+        requires std::random_access_iterator<it_t>
+    //!\endcond
+    {
+        window_left += n;
+        get_minimiser();
+        return operator*();
+    }
+
+private:
+    //!brief Iterator to last element in range.
+    sentinel_t last_elem;
```

Comment:
Since the name `last_elem` could also refer to the last elem in the window
```suggestion
    sentinel_t urange_end;
```